common/views/components/MediaObjectBase/MediaObjectBase.tsx

```diff
@@ -0,0 +1,151 @@
+  DateInfo: ReactElement<typeof DateRange> | ReactElement<typeof EventDateRange> | null,
+  StatusIndicator: ReactElement<typeof StatusIndicator> | null,
+  ExtraInfo?: ReactNode | null,
+  xOfY: { x: number, y: number },
+  OverrideImageWrapper?: ReactNode,
+  OverrideTextWrapper?: ReactNode,
+  OverrideTitleWrapper?: ReactNode,
+  onClick?: () => void,
+};
+
+const BaseImageWrapper = styled.div.attrs(props => ({
+  className: grid({ s: 3, m: 3, l: 3, xl: 3 }),
+}))``;
+
+const BaseTitleWrapper = styled.div.attrs(props => ({
+  className: classNames({
+    'card-link__title': true,
+    [font('wb', 5)]: true,
+  })
+}))``;
+
+export type TextWrapperProp = {
+  hasImage: boolean
+};
+
+// Ability to add custom prop types in TS and styled components
+const BaseTextWrapper = styled.div.attrs<TextWrapperProp>(props => {
+  if (props.hasImage) {
+    return {
+      className: grid({ s: 9, m: 9, l: 9, xl: 9 }),
+    };
+  }
+  return {
+    className: grid({ s: 12, m: 12, l: 12, xl: 12 }),
+  };
+  
+})<TextWrapperProp>``;
+
+const MediaObjectBase = ({
+  url,
+  title,
+  labels,
+  description,
+  urlOverride,
+  extraClasses,
+  partNumber,
+  color,
+  Image,
+  DateInfo,
+  StatusIndicator,
+  ExtraInfo,
+  xOfY,
+  OverrideImageWrapper,
+  OverrideTextWrapper,
+  OverrideTitleWrapper,
+  onClick,
+} : Props) => {
+  const { x, y } = xOfY;
+
+  const ImageWrapper:any = OverrideImageWrapper ? OverrideImageWrapper : BaseImageWrapper;
+  const TextWrapper:any = OverrideTextWrapper ? OverrideTextWrapper : BaseTextWrapper;
+  const TitleWrapper:any = OverrideTitleWrapper ? OverrideTitleWrapper : BaseTitleWrapper;
+
+  const descriptionIsString = typeof description === 'string';
+
+  return (
+    <Space
+      v={{
+        size: 'l',
+        properties: [
+          'padding-top',
+          x === y ? undefined : 'padding-bottom',
+        ].filter(Boolean),
+      }}
+      as={url ? 'a' : 'div'}
+      href={urlOverride || url}
+      className={conditionalClassNames({
+        grid: true,
+        'card-link': Boolean(url),
+        [extraClasses || '']: Boolean(extraClasses),
+      })}
+      onClick={() => {
+        if (onClick) {
+          onClick();
+        }
+      }}
+    >
+      <ImageWrapper hasImage={Boolean(Image)}> 
+        {Image}
+      </ImageWrapper>
+      <TextWrapper hasImage={Boolean(Image)} >
+      {labels.labels.length > 0 && (
+          <Space
+            v={{ size: 's', properties: ['margin-bottom'] }}
+            className="flex"
+          >
+            <LabelsList {...labels} />
+          </Space>
+        )}
+        {partNumber && (
+          <PartNumberIndicator number={partNumber} color={color} />
+        )}
+        <TitleWrapper>
+          {title}
+        </TitleWrapper>
+        {DateInfo}
+        {StatusIndicator}
+        {ExtraInfo}
+        {description && 
+          <div className={`spaced-text ${!descriptionIsString && font('hnl', 5)}`}>
```

Comment:
we tend to handle conditional classes with the [`classNames()` helper](https://github.com/wellcomecollection/wellcomecollection.org/blob/master/common/utils/classnames.js#L60)

```
<div className={classNames({
  'spaced-text': true,
  [font('hnl', 5')]: !descriptionIsString,
})>
```

Reply:
nice to know this util can do that.. added this in